File: learnAPI/5a.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "https://api.zippopotam.us/us/10006"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- as.data.frame(data)
    print(head(df))
    cat('\n\n\n')

    entry <- data
    data <- data$places

    df <- data.frame(
        post_code = entry$`post code`,
        country = entry$country,
        country_code = entry$`country abbreviation`,
        place = data$`place name`,
        state = data$state,
        state_abbreviation = data$`state abbreviation`,
        longitude = data$longitude,
        latitude = data$latitude
    )
    print(df)
}

File: learnAPI/5a.py
import pandas as pd
import requests

url = 'https://api.zippopotam.us/us/33162'

response = requests.get(url)

if response.status_code == 200:
    data = response.json()
    results = data['places']
    
    df_list = []
    
    for entry in results:
        df_entry = pd.DataFrame({
            'post_code' : [data['post code']],
            'country' : [data['country']],
            'country_abbreviation' : [data['country abbreviation']],
            'place' : entry['place name'],
            'state' : entry['state'],
            'state_abbreviation' : entry['state abbreviation'],
            'longitude' : entry['longitude'],
            'latitude' : entry['latitude']
        })
        df_list.append(df_entry)
    
    df = pd.concat(df_list, ignore_index=True)
    print(df)
else:
    print("Failed to retrieve data:", response.status_code)